lib/rubocop/cop/cop.rb

```diff
@@ -183,8 +189,19 @@ def relevant_file?(file)
           !file_name_matches_any?(file, 'Exclude', false)
       end
 
+      def style_guide_url
+        url = cop_config && cop_config['StyleGuide']
+        (url.nil? || url.empty?) ? nil : url
+      end
+
       private
 
+      def annotate_message(message)
+        message = "#{name}: #{message}" if display_cop_names?
+        message = "#{message} (#{style_guide_url})" if display_style_guide?
+        message
+      end
```

Comment:
@bquorning I'm not sure exactly what you mean by 'change' but my implementation doesn't mutate the `message` argument, if that's your concern.